binutil/sizedfile.go

```diff
@@ -17,6 +17,7 @@ type SizedFile struct {
 
 func (r *SizedFile) Read(p []byte) (n int, err error) { return r.s.Read(p) }
 func (r *SizedFile) Size() int64                      { return r.s.Size() }
+func (r *SizedFile) AlignedSize() int64               { return Align(r.s.Size()) }
```

Comment:
[optional] I'm tempted to try introducing just a single helper struct:

```
struct Aligned { SizedReader }
func (a *Aligned) Padding() []byte {
	s := a.Size()
	aligned := (s-1)&^7 + 8
	pad := [8]byte{}
	return pad[aligned-s]
}
```

and special-casing it where needed (with `len(a.Padding())` when just size is needed). Then wrapping any objects that need to be aligned at the place where they're *used* - instead of adding .AlignedSize() to SizedFile and __GRPICONDIR. I would assume it's the *use site* that determines if we need to align a particular object, instead of alignment being the object's intrinsic quality. But I'm more than happy to take a stab at this refactoring once you're done with your awesome work and once you've added some tests. I'm thus adding this comment more as a note to myself to not forget about that idea later.

Reply:
Yes, I agree too, my fix is illogical for sure.
I've never taken time to fully understand the code, as I am uncomfortable with its generic nature (and I'm new to Go).
It is still unclear where the call site is to me, so I'd rather let you fix that :)

(should I click on "Resolve conversation" ? I'm new to github too!)